### PR TITLE
SC-2555: Readme updates

### PR DESCRIPTION
**Pull request by sivasamyk**
The Readme provide shot intro to Sematext App Agent, How to build the agent package, install and setup new app. It also covers how to adjust agent properties to send metrics to InfluxDB. Other details regarding agent like running modes, config etc will present in www docs and provided a link for the same.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,280 +1,70 @@
...
-        source: func:StringEquals(upstream.server.state,up)
-        type: gauge
-        label: state up
-        description: Server is up
-```
+After cloning the repo, executing `build.sh` will build the packages for multiple Linux distributions.
 
-- TimeSince - Returns time in ms passed since event was accured. Simple calculation of now() - time_contained_in_metric_param
+### Setup
+The packages can be installed using OS specific package manager like dpkg, yum, etc. Once installed a new App can be
```
Comment (bsmid) [inline README.md]: I think we need some quick explanation of what is an "App".

File: README.md [hunk 2]
```diff
@@ -1,280 +1,70 @@
...
+sudo bash /opt/spm/bin/setup-spm  \
+    --app-token <app-token>   \
+    --app-type jvm  \
+    --agent-type standalone \
+    --jmx-params '-Dspm.remote.jmx.url=localhost:3000'
 ```
 
-### Service type specific built-in functions:
-- com.sematext.spm.client.hadoop.CalculateNumNodes - hadoop specific. The only param is metric name mapped to hadoop jmx attribute LiveNodes.
+`<app-token>` - App Token is should point to Sematext App Token if you are sending metrics to Sematext. For other Influx endpoints
```
Comment (bsmid) [inline README.md]: Remove "is". "to Sematext" -> "to Sematext Cloud"?

File: README.md [hunk 3]
```diff
@@ -1,280 +1,70 @@
...
-      - name: nn.nodes.live
-        source: func:com.sematext.spm.client.hadoop.CalculateNumNodes(nodes.live)
-        type: long_gauge
-        label: live nodes
-```
+Visit [Sematext Documentation](https://sematext.com/docs/monitoring/spm-client/) for more info on how to setup and 
+configure agent to ship metrics.
 
-- com.sematext.spm.client.solr.CalculateCoreNumFiles - solr specific, calculates number of files for some solr core (can't use universal CountFiles function due to Solr internal specifics)
+By default the agent sends the collected metrics to Sematext. You can configure different Influx compatible destination
```
Comment (bsmid) [inline README.md]: "to Sematext" -> "to Sematext Cloud"?

File: README.md (hunk 1 above)
Comment (otisg) [reply README.md]: @megastef will have a What's an App video ready soon.
@goranbajic worked on a Monitoring Guide (https://sematext.com/docs/guide/monitoring-guide/) where I now hoped to see the concept of an App explained, but it looks like we didn't include that (yet?).

File: README.md (hunk 3 above)
Comment (otisg) [reply README.md]: I think "Sematext" is OK, too.  We use that a lot elsewhere.

different => a different

File: README.md [hunk 4]
```diff
@@ -13,7 +13,8 @@ Currently supported data sources are:
 * SQL
 
 Sematext App Agent uses Influx Line Protocol to ship the metrics. The metrics collected by the agent can be shipped to
-any Influx Line Protocol complaint endpoints like InfluxDB. You can also add support for other output formats like HTTP, Graphite, etc.
+any Influx Line Protocol complaint endpoints like InfluxDB. In future, you can also add support for other output formats
```
Comment (otisg) [inline README.md]: complaint => compliant
In future, you can also add => In the future, we will add

File: README.md [hunk 5]
```diff
@@ -38,7 +47,7 @@ setup by running `setup-spm` command. For example, to setup monitoring for a JVM
 ```bash
 -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=3000 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false
 ```
-VM arguments to the Java startup process. Then you can setup new app by running the following command:
+to startup arguments of Java process you wish to monitor. Then you can setup new App by running the following command:
```
Comment (otisg) [inline README.md]: setup => set up (noun vs. verb)
new App => a new App

File: README.md [hunk 6]
```diff
@@ -48,7 +57,7 @@ sudo bash /opt/spm/bin/setup-spm  \
     --jmx-params '-Dspm.remote.jmx.url=localhost:3000'
 ```
 
-`<app-token>` - App Token is should point to Sematext App Token if you are sending metrics to Sematext. For other Influx endpoints
+`<app-token>` - App Token should point to Sematext App Token if you are sending metrics to Sematext. For other Influx endpoints
 you can specify a hexa-decimal value with format `xxxxxxxx--xxxx-xxxx-xxxx-xxxxxxxxxxxx`. e.g. `d0add28a-0a0f-46b2-9e1e-4928db5200e7`
```
Comment (otisg) [inline README.md]: hexa-decimal => hexadecimal

**Pull request merged**